src/pulsar_producer.erl

```diff
@@ -571,6 +608,7 @@ handle_response({send_receipt, Resp = #{sequence_id := SequenceId}}, State) ->
               FromsToMessages),
             InflightCalls = InflightCalls0 - BatchSize,
             pulsar_metrics:inflight_set(State, InflightCalls),
+            self() ! #maybe_send_to_pulsar{},
```

Comment:
use gen_statem actions in the return instead ?

Reply:
:checkered_flag: